code/game/machinery/doors/poddoor.dm

```diff
@@ -97,11 +97,46 @@
 	sleep(5)
 
 	operating = 0
-	return
+	return 1
 
-/obj/machinery/door/poddoor/multi_tile // Whoever wrote the old code for multi-tile spesspod doors needs to burn in hell.
+ // Whoever wrote the old code for multi-tile spesspod doors needs to burn in hell. - Unknown
+ // Wise words. - Bxil
+/obj/machinery/door/poddoor/multi_tile
 	name = "large pod door"
 	layer = CLOSED_DOOR_LAYER
+	var/idir 		//Initial Direction. For some reason upon spawning these doors always turn to south.
+					//We must force them to their corret direction, so we store it here.
+					//Bad solution, I know. If you know how to fix please do. - Bxil
+
+/obj/machinery/door/poddoor/multi_tile/New()
+	idir = dir
+	. = ..()
+	apply_opacity_to_my_turfs(opacity)
+
+/obj/machinery/door/poddoor/multi_tile/open()
+	if (..())
+		apply_opacity_to_my_turfs(opacity)
+
+
+/obj/machinery/door/poddoor/multi_tile/close()
+	if (..())
+		apply_opacity_to_my_turfs(opacity)
+
+/obj/machinery/door/poddoor/multi_tile/Destroy()
+	apply_opacity_to_my_turfs(0)
+	return ..()
+
+//Poddoors don't turn opaque automatically. Fear not, here is the fix.
+//A bad fix. Which needs exorcism. (Disclaimer: exorcism in not in the code YET. Upon being coded, please apply immediately.)
+/obj/machinery/door/poddoor/multi_tile/proc/apply_opacity_to_my_turfs(var/new_opacity)
+	var/numsteps = 1
+	var/turf/current_turf = get_turf(src)
+	while(numsteps <= width)
+		current_turf.opacity = new_opacity

```

Comment:
Add `current_turf.has_opaque_atom = TRUE` and `currrent_turf.reconsider_lights()`